src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5285,6 +5285,48 @@ protected sealed override ISymbol GetEnclosingSymbolCore(int position, Cancellat
             return this.GetEnclosingSymbol(position, cancellationToken);
         }
 
+        private protected sealed override IImportScope GetImportScopeCore(int position, CancellationToken cancellationToken)
+        {
+            position = CheckAndAdjustPosition(position);
+            var binder = GetEnclosingBinder(position);
+            return ConvertToImportScope(binder?.ImportChain);
+
+            static IImportScope ConvertToImportScope(ImportChain chain)
+            {
+                // Skip by any empty items in the chain.
+                while (chain != null && chain.Imports.IsEmpty)
+                    chain = chain.ParentOpt;
+
+                // If we reached the end, there's nothing to return
+                if (chain == null)
+                    return null;
+
+                var imports = chain.Imports;
+                Debug.Assert(!imports.IsEmpty);
+
+                // Try to create a node corresponding to the imports of the next higher binder scope. Then create the
+                // node corresponding to this set of imports and chain it to that.
+                return new SimpleImportScope(
+                    ConvertToImportScope(chain.ParentOpt),
+                    ConvertAliases(imports),
+                    imports.ExternAliases.SelectAsArray(static e => e.Alias.GetPublicSymbol()),
```

Comment:
>ExternAliases [](http://example.com/codeflow?start=28&length=13)

Is the order in this array deterministic?

Reply:
Currently, the docs for the feature state that no order is guaranteed for the elements of any of these properties.  We can def discuss in the next meeting.